scripts/pi-hole/js/messages.js

```diff
@@ -57,6 +57,24 @@ function renderMessage(data, type, row) {
         "</pre> to get the group configuration for this client."
       );
 
+    case "HOSTNAME":
+      var hint = "";
+      for (var i = 0; i < row.blob2 + row.message.length + 2; i++) hint += " ";
+      return (
+        "Host names contains invalid character <code> " +
+        decodeURIComponent(escape(row.blob1))[row.blob2] +
+        "</code>:<pre>" +
+        hint +
+        "↓\n" +
```

Comment:
Might be worth looking into using the HTML entities for the symbols here. We don't use Unicode symbols in JS AFAICT.

Reply:
Can HTML entities be used inside `<pre>`?

Reply:
If it's not escaped it's not valid IIRC. But anyway just give it a go.